Sources/ApolloCodegenLib/IR/ScopedSelectionSetHashable.swift

```diff
@@ -13,38 +13,21 @@ protocol ScopedSelectionSetHashable {
   var hashForSelectionSetScope: String { get }
 }
 
-extension CompilationResult.Selection: ScopedSelectionSetHashable {
-  var hashForSelectionSetScope: String {
-    switch self {
-    case let .field(selection as ScopedSelectionSetHashable),
-      let .inlineFragment(selection as ScopedSelectionSetHashable),
-      let .fragmentSpread(selection as ScopedSelectionSetHashable):
-      return selection.hashForSelectionSetScope
-    }
-  }
-}
-
-extension CompilationResult.Field: ScopedSelectionSetHashable {
-  var hashForSelectionSetScope: String {
-    return responseKey
-  }
-}
-
-extension ASTField: ScopedSelectionSetHashable {
+extension IR.SelectionSet: ScopedSelectionSetHashable {
+#warning("What if there is a field with the same name as a type? Do we get a conflict? Write a test for this.")
```

Comment:
Adding a comment as a flag to check that we're testing this later (and for @AnthonyMDev to put a link to the commit with that test 😇)